app/models/concerns/patient_date_validator.rb

```diff
@@ -8,9 +8,11 @@ def validate(record)
     year_start = Date.new(2020, 1, 1)
     month_ahead = 30.days.from_now.to_date
     month_behind = 30.days.ago.to_date
-    validate_between_dates(record, :date_of_birth, Date.new(1900, 1, 1), Time.now.to_date) if record.date_of_birth_changed?
+    # This is to handle the edge case when certain time zones such as Samoa are on the new day before UTC
+    earliest_tz_today = Time.now.in_time_zone(ActiveSupport::TimeZone.all.last).to_date
+    validate_between_dates(record, :date_of_birth, Date.new(1900, 1, 1), earliest_tz_today) if record.date_of_birth_changed?
     validate_between_dates(record, :last_date_of_exposure, year_start, month_ahead) if record.last_date_of_exposure_changed?
-    validate_between_dates(record, :symptom_onset, year_start, month_ahead) if record.symptom_onset_changed?
+    validate_between_dates(record, :symptom_onset, year_start, earliest_tz_today) if record.symptom_onset_changed?
```

Comment:
Although I think it makes sense to add the validation to the model for symptom onset, it does scare me a little to go ahead and add all these validations as part of a ticket that does not really require them, since we pushed this off for further testing, and I'm not sure if that further testing has happened yet.

Reply:
I briefly talked to @holmesie about this [here](https://mitrecorp.slack.com/archives/GSV87B6BB/p1626979129029700) and he said we just need to check our data in prod which we can do before deploying this. Was there any other kind of testing you had in mind for this?

Reply:
My understanding from @holmesie was that we were fine with including these model level validations, but only for symptom onset date. There is the risk of data on prod being wrong, but there is also a risk of the model validations aligning incorrectly with the front end validation, which could cause some pretty bad bugs. And the risk of that goes up if we add more validations.

I clarified with @holmesie earlier today and he said he was intending we just do symptom onset, so how would you feel about leaving the other validations commented out for this PR?

Reply:
https://github.com/SaraAlert/SaraAlert/pull/1018/commits/0afe132d882ccfcc3c206147638372a48b99c4b0